test/rebar_install_deps_SUITE.erl

```diff
@@ -201,9 +202,9 @@ mdeps(m_circular2) ->
      {error, {rebar_prv_install_deps, {cycles, [[<<"B">>,<<"C">>]]}}}};
 mdeps(m_circular3) ->
     %% Spot the circular dep due to being to low in the deps tree
-    %% but as a source dep, taking precedence over packages
+    %% but as a source dep, taking precedence over packages at the same level
     {[{"B", [{"C", "2", [{"B", []}]}]},
-      {"c", "1", [{"d",[]}]}],
+      {"D", [{"c", "1", [{"d",[]}]}]}],
```

Comment:
We should keep it at a higher level. Let's say:
```erlang
{[{"B", [{"Z", [{"C", "2", [{"B", []}]}]}]},
  {"D", [{"c", "1", [{"d",[]}]}]}],
 [],
 {error, {rebar_prv_install_deps, {cycles, [[<<"B">>,<<"C">>]]}}}};
```
We already test precedence rules at similar levels, but this one of the few that may properly check for "only the top level pkg wins" indirectly, and forces us to check that a lower source dep takes precedence over a higher (but not top-level) package.